components/builder-sessionsrv/src/data_store.rs

```diff
@@ -59,13 +57,27 @@ impl DataStore {
         Ok(())
     }
 
-    fn row_to_account(&self, row: postgres::rows::Row) -> sessionsrv::Account {
-        let mut account = sessionsrv::Account::new();
-        let id: i64 = row.get("id");
-        account.set_id(id as u64);
-        account.set_email(row.get("email"));
-        account.set_name(row.get("name"));
-        account
+    pub fn account_find_or_create(
+        &self,
+        msg: &sessionsrv::AccountFindOrCreate,
+    ) -> SrvResult<sessionsrv::Account> {
+        // JW TODO: optimize this to be one database call
+        let conn = self.pool.get(msg)?;
+        let rows = conn.query(
+            "SELECT * FROM get_account_by_name_v1($1)",
+            &[&msg.get_name()],
+        )?;
+        if rows.len() != 0 {
+            let row = rows.get(0);
+            Ok(self.row_to_account(row))
+        } else {
+            let rows = conn.query(
+                "SELECT * FROM select_or_insert_account_v1($1, $2)",
+                &[&msg.get_name(), &msg.get_email()],
```

Comment:
Would this call by itself get you what you want?

Reply:
Ummm, why yes! Hahaha. I actually wrote a TODO above this reminding myself to write this function that already exists that I'm already calling :). Thanks for the catch man